File: shared/src/commonMain/kotlin/com/vocapop/domain/models/UserCard.kt
package com.vocapop.domain.models

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class UserCard(
    val id: String,
    val userId: String,
    val cardId: String,
    val easeFactor: Float = 2.5f,
    val intervalDays: Int = 1,
    val repetitions: Int = 0,
    val nextReviewDate: LocalDateTime,
    val lastReviewDate: LocalDateTime? = null
)

@Serializable
data class CardWithProgress(
    val card: Card,
    val progress: UserCard? = null
)

File: backend/app/Models/UserCard.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class UserCard extends Model
{
    use HasFactory, HasUuids;

    protected $fillable = [
        'user_id',
        'card_id',
        'ease_factor',
        'interval_days',
        'repetitions',
        'next_review_date',
        'last_review_date',
    ];
    
    protected $casts = [
        'next_review_date' => 'datetime',
        'last_review_date' => 'datetime',
    ];
    
    public function user()
    {
        return $this->belongsTo(User::class);
    }
    
    public function card()
    {
        return $this->belongsTo(Card::class);
    }
}
